app/script/cryptography/CryptographyMapper.js

```diff
@@ -0,0 +1,372 @@
+        info: {
+          name: original.name,
+          nonce: event_nonce,
+        },
+        meta: this._map_asset_meta_data(original),
+      },
+      type: z.event.Client.CONVERSATION.ASSET_META,
+    };
+  }
+
+  _map_asset_preview(preview, event_id) {
+    return {
+      data: {
+        id: event_id,
+        otr_key: new Uint8Array(preview.remote.otr_key !== null ? preview.remote.otr_key.toArrayBuffer() : undefined),
+        sha256: new Uint8Array(preview.remote.sha256 !== null ? preview.remote.sha256.toArrayBuffer() : undefined),
+        key: preview.remote.asset_id,
+        token: preview.remote.asset_token,
+      },
+      type: z.event.Client.CONVERSATION.ASSET_PREVIEW,
+    };
+  }
+
+  _map_asset_uploaded(uploaded, event_id) {
+    return {
+      data: {
+        id: event_id,
+        otr_key: new Uint8Array(uploaded.otr_key !== null ? uploaded.otr_key.toArrayBuffer() : undefined),
+        sha256: new Uint8Array(uploaded.sha256 !== null ? uploaded.sha256.toArrayBuffer() : undefined),
+        key: uploaded.asset_id,
+        token: uploaded.asset_token,
+      },
+      type: z.event.Client.CONVERSATION.ASSET_UPLOAD_COMPLETE,
+    };
+  }
+
+  _map_cleared(cleared) {
+    return {
+      conversation: cleared.conversation_id,
+      data: {
+        cleared_timestamp: cleared.cleared_timestamp.toString(),
+      },
+      type: z.event.Backend.CONVERSATION.MEMBER_UPDATE,
+    };
+  }
+
+  _map_confirmation(confirmation) {
+    return {
+      data: {
+        message_id: confirmation.message_id,
+        status: (() => {
+          switch (confirmation.type) {
+            case z.proto.Confirmation.Type.DELIVERED:
+              return z.message.StatusType.DELIVERED;
+            case z.proto.Confirmation.Type.READ:
+              return z.message.StatusType.SEEN;
+          }
+        })(),
+      },
+      type: z.event.Client.CONVERSATION.CONFIRMATION,
+    };
+  }
+
+  _map_deleted(deleted) {
+    return {
+      data: {
+        message_id: deleted.message_id,
+      },
+      type: z.event.Client.CONVERSATION.MESSAGE_DELETE,
+    };
+  }
+
+  _map_edited(edited, event_id) {
+    const mapped = this._map_text(edited.text, event_id);
+    mapped.data.replacing_message_id = edited.replacing_message_id;
+    return mapped;
+  }
+
+  _map_ephemeral(generic_message, event) {
+    const millis_as_number = generic_message.ephemeral.expire_after_millis.toNumber();
+    generic_message.ephemeral.message_id = generic_message.message_id;
+    const embedded_message = this._map_generic_message(generic_message.ephemeral, event);
+    embedded_message.ephemeral_expires = z.ephemeral.timings.map_to_closest_timing(millis_as_number);
+    return embedded_message;
+  }
+
+  /*
+  Unpacks a specific generic message which is wrapped inside an external generic message.
+
+  @note Wrapped messages get the 'message_id' of their wrappers (external message)
+  @param {z.proto.GenericMessage} external Generic message of type 'external'
```

Comment:
- Asterisk style needs to be:
```js
      /**
       * 
       * 
       */
```
- We use hyphens before the description of a param, as shown in: http://usejsdoc.org/tags-param.html#names-types-and-descriptions